cmd/cloud-run/signifysecretrotator/signify/client.py

```diff
@@ -0,0 +1,123 @@
+
+            old_cert_file.write(certificate)
+            old_cert_file.flush()
+
+            old_key_file.write(private_key)
+            old_key_file.flush()
+
+            access_token_response = requests.post(
+                self.token_url,
+                cert=(old_cert_file.name, old_key_file.name),
+                data={
+                    "grant_type": "client_credentials",
+                    "client_id": self.client_id,
+                },
+                timeout=30,
+            )
+
+            if access_token_response.status_code != 200:
+                raise requests.HTTPError(
+                    f"Got not-success status code {access_token_response.status_code}",
+                    response=access_token_response,
+                )
+
+            decoded_response = access_token_response.json()
+
+            if "access_token" not in decoded_response:
+                raise ValueError(
+                    f"Got unexpected response structure: {decoded_response}"
+                )
+
+            return decoded_response["access_token"]
+
+    def fetch_new_certificate(
+        self, cert_data: bytes, private_key: rsa.RSAPrivateKey, access_token: str
+    ):
+        """Fetch new certificates from given certificate service"""
+
+        csr = self._prepare_csr(cert_data, private_key)
+
+        crt_create_payload = self._prepare_cert_request_paylaod(csr)
+
+        cert_create_response = requests.post(
+            self.certificate_service_url,
+            headers={
+                "Authorization": f"Bearer {access_token}",
+                "Content-Type": "application/json",
+                "Accept": "application/json",
+            },
+            data=crt_create_payload,
+            timeout=10,
+        )
+
+        if cert_create_response.status_code != 200:
+            raise requests.HTTPError(
+                f"Got un-success statsu code {cert_create_response.status_code}"
+            )
+
+        decoded_response = cert_create_response.json()
+
+        if (
+            "certificateChain" not in decoded_response
+            or "value" not in decoded_response["certificateChain"]
+        ):
+            raise ValueError(
+                f"Cannot issue new certifacte, invalid response format: {decoded_response}"
+            )
+
+        pkcs7_certs = decoded_response["certificateChain"]["value"].encode()
+
+        return pkcs7.load_pem_pkcs7_certificates(pkcs7_certs)
+
+    def _prepare_cert_request_paylaod(self, csr: x509.CertificateSigningRequest):
+        return json.dumps(
+            {
+                "csr": {
+                    "value": csr.public_bytes(serialization.Encoding.PEM).decode(
+                        "utf-8"
+                    )
+                },
+                "validity": {"value": 7, "type": "DAYS"},
+                "policy": "sap-cloud-platform-clients",
+            }
+        )
+
+    def _prepare_csr(self, cert_data: bytes, private_key: rsa.RSAPrivateKey):
```

Comment:
missing type annotations in whole file. please revisit all code

Reply:
Added types, even when I not agree with "missing type annotations in whole file" when on commented lien there is type annotation.